### DISCUSSION TITLE
How can I custom the validation rule?

### DISCUSSION THREAD

**Discussion by linhchelsea** [adonisjs/core#2615] [General]
I want to validate a number that must be an integer. But I can not find that rule in this document https://docs.adonisjs.com/reference/validator/rules/alpha

**Comment by mastermunj** [comment]
You can create your own custom validation rules.

`start/validationRules.ts`
```ts
validator.rule('integer', (value: number, _, { pointer, arrayExpressionPointer, errorReporter }) => {
  const isValid = Number.isInteger(value);

  if (!isValid) {
    errorReporter.report(pointer, 'integer', 'Enter a valid integer value', arrayExpressionPointer);
  }
});
```

`contracts/validator.ts`
```ts
export interface Rules {
  integer(): Rule;
}
```

Now you can use the rule as 
```ts
public schema = schema.create({
  amount: schema.number([rules.integer()]),
});
```

**Reply by linhchelsea** [reply]
Fantastic!

It works for me

Thank you so much <3

**Comment by arthurcvm** [comment]
@thetutlage The custom rules page is not in the new docs? I need consult this.
